pkg/networking/cilium/reconciler/reconciler.go

```diff
@@ -103,12 +110,20 @@ func (r *Reconciler) Reconcile(ctx context.Context, logger logr.Logger, client c
 
 	if upgradeInfo.VersionUpgradeNeeded() {
 		logger.Info("Cilium upgrade needed", "reason", upgradeInfo.Reason())
-
-		if result, err := r.upgrade(ctx, logger, client, installation, spec); err != nil {
+		result, err := r.upgrade(ctx, logger, client, installation, spec)
+		if err != nil {
 			return controller.Result{}, err
-		} else if result.Return() {
+		}
+
+		// When there is no error but a requeue has been returned, then a version upgrade is in progress
+		if !result.Result.IsZero() {
+			conditions.MarkFalse(spec.Cluster, anywherev1.DefaultCNIConfiguredCondition, anywherev1.DefaultCNIUpgradeInProgressReason, clusterv1.ConditionSeverityInfo, "Cilium version upgrade needed")
+		}
+
+		if result.Return() {
```

Comment:
this seems redundant
why not keep the code as it was and just add the markFalse just before the return?

Reply:
If `result.Return()` just means `r.Result != nil`, then that means it can possible that it does not contain a re-queue request. So, here I wanted to make sure that there was a re-queue, without nesting another if statement in that block that checks for `result.Return()`.